### PR TITLE
parser - Add handling for carriage returns in wiki

### PR DESCRIPTION
**Pull request by josepht**
LP:#1605541

### CODE REVIEW

File: snapcraft/internal/parser.py [hunk 1]
```diff
@@ -240,9 +241,9 @@ def _process_index(output):
 
     # split the wiki into an array of entries to allow the parser to
     # proceed when invalid yaml is found.
-    entries = output.split(b'\n---\n')
+    entries = re.split(r'\r*\n---\r*\n', output.decode(), flags=re.MULTILINE)
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: I would read line by line in the decoded already and looks for an empty newline it might be less error prone.

Comment (josepht) [inline snapcraft/internal/parser.py]: I've updated the PR to do linewise parsing of the wiki using '---' lines as delimiters.

File: snapcraft/internal/parser.py [hunk 2]
```diff
@@ -229,40 +229,52 @@ class Options:
...
-                parts_list, after_parts = _process_entry(data)
-            except InvalidEntryError as e:
-                logger.warning('Invalid wiki entry: {}'.format(e))
-                continue
-
-            if is_valid_parts_list(parts_list, after_parts):
-                master_parts_list.update(parts_list)
+    entry = ""
+    for line in output.decode().splitlines():
+
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: Remove empty line

File: snapcraft/internal/parser.py [hunk 3]
```diff
@@ -229,40 +229,52 @@ class Options:
...
+        if line == "---":
+            if entry != "":
+                _process_wiki_entry(entry, master_parts_list)
+                entry = ""
+        else:
+            entry = "\n".join([entry, line])
+
+    if entry != "":
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: Just `if entry`

File: snapcraft/internal/parser.py [hunk 4]
```diff
@@ -229,40 +229,52 @@ class Options:
...
-                logger.warning('Invalid wiki entry: {}'.format(e))
-                continue
-
-            if is_valid_parts_list(parts_list, after_parts):
-                master_parts_list.update(parts_list)
+    entry = ""
+    for line in output.decode().splitlines():
+
+        if line == "---":
+            if entry != "":
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: Just `if entry`

Comment (sergiusens) [inline snapcraft/internal/parser.py]: just if entry

File: snapcraft/internal/parser.py [hunk 5]
```diff
@@ -229,40 +229,52 @@ class Options:
...
+                _process_wiki_entry(entry, master_parts_list)
+                entry = ""
+        else:
+            entry = "\n".join([entry, line])
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: use single quotes

File: snapcraft/internal/parser.py [hunk 6]
```diff
@@ -229,40 +229,52 @@ class Options:
...
-            except InvalidEntryError as e:
-                logger.warning('Invalid wiki entry: {}'.format(e))
-                continue
-
-            if is_valid_parts_list(parts_list, after_parts):
-                master_parts_list.update(parts_list)
+    entry = ""
+    for line in output.decode().splitlines():
+
+        if line == "---":
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: Use single quotes please

File: snapcraft/internal/parser.py [hunk 7]
```diff
@@ -229,40 +229,52 @@ class Options:
...
-
-            try:
-                parts_list, after_parts = _process_entry(data)
-            except InvalidEntryError as e:
-                logger.warning('Invalid wiki entry: {}'.format(e))
-                continue
-
-            if is_valid_parts_list(parts_list, after_parts):
-                master_parts_list.update(parts_list)
+    entry = ""
```
Comment (sergiusens) [inline snapcraft/internal/parser.py]: Use single quotes please

**Comment by josepht** [comment]
@sergiusens I've fixed the issues from all of your review comments.

**Pull request merged**